crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -44,6 +48,7 @@ pub struct SettlementEncoder {
     // would make the trait not be object safe which prevents using it through `dyn`.
     // TODO: Can we fix this in a better way?
     execution_plan: Vec<Arc<dyn Interaction>>,
+    pre_interactions: Vec<Arc<InteractionData>>,
```

Comment:
nit: I wouldn't additionally introduce an `Arc` here. If we want to use shared pointers for the vector, I would prefer:
```rust
    pre_interactions: Arc<Vec<InteractionData>>,
```

That being said, I don't think this is necessary at all and would use:
```rust
    pre_interactions: Vec<InteractionData>,
```

Note that we need the `Arc` above because we are using a "trait-object" and not a concrete type (here, a struct).